src/Account/Controller/UserController.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  CreateUserResponse,
+  GetUserGeoInfoResponse,
+  GetUserNearByFriendResponse,
+} from './Response/UserControllerResponse';
+
+@ApiTags('User Service')
+@Controller('/user')
+export default class UserController {
+  @Inject() private useService: UserService;
+  @Inject() private userGeoService: UserGeoService;
+
+  @ApiOperation({
+    description: 'Create New User',
+    summary: 'Create User',
+  })
+  @ApiCreatedResponse({
+    status: HttpStatus.OK,
+    description: 'succeeded create user',
+    type: CreateUserResponse,
+  })
+  @ApiBadRequestResponse(SwaggerBadRequestResponse())
+  @HttpCode(HttpStatus.CREATED)
+  @Post('')
+  public createUser(@Body() createUserDto: CreateUserDto) {
+    return this.useService.createUser(createUserDto);
+  }
+
+  @ApiOperation({
+    description: 'Delete User',
+    summary: 'Delete User',
+  })
+  @ApiNoContentResponse()
+  @ApiBadRequestResponse(SwaggerBadRequestResponse())
+  @HttpCode(HttpStatus.NO_CONTENT)
+  @Delete('/:id')
+  public deleteUser(@Param('id', new ObjectIdValidationPipe()) id: string) {
+    return this.useService.deleteUserWithId(id);
+  }
+
+  @ApiOperation({
+    description: 'Get User Data',
+    summary: 'Get User Data',
+  })
+  @ApiBadRequestResponse(SwaggerBadRequestResponse())
+  @HttpCode(HttpStatus.OK)
+  @Get('/:id')
+  public getUser(@Param('id', new ObjectIdValidationPipe()) id: string) {
+    return this.useService.getUserWithId(id);
+  }
+
+  @ApiOperation({
+    description: 'Update User Data',
+    summary: 'Update User Data',
+  })
+  @ApiBadRequestResponse(SwaggerBadRequestResponse())
+  @HttpCode(HttpStatus.OK)
+  @Patch('/:id')
+  public updateUserBaseInfo(
+    @Param('id', new ObjectIdValidationPipe()) userId: string,
+    @Body() updateUserDto: UpdateUserDto,
+  ) {
+    return this.useService.updateUser(userId, updateUserDto);
+  }
+
+  @ApiOperation({
+    description: 'Update User Geo Data',
+    summary: 'Update User Geo Data',
+  })
+  @ApiBadRequestResponse(SwaggerBadRequestResponse())
+  @HttpCode(HttpStatus.OK)
+  @Patch('/:id/geo')
+  public updateUserGeoInfo(
+    @Param('id', new ObjectIdValidationPipe()) id: string,
+    @Body() updateUserGeoInfoDto: UpdateUserGeoInfoDto,
+  ) {
+    return this.userGeoService.updateUserGeoInfo(id, updateUserGeoInfoDto);
+  }
+
+  @ApiOperation({
+    description: 'get geo info',
+    summary: 'Get User Geo info',
+  })
+  @ApiOkResponse({
+    description: 'ok',
+    type: GetUserGeoInfoResponse,
+  })
+  @ApiBadRequestResponse(SwaggerBadRequestResponse())
+  @Get('/:id/geo')
+  async getUserGeoInfo(@Param('id', new ObjectIdValidationPipe()) id: string) {
+    return this.userGeoService.getGeoInfoWithId(id);
+  }
+
+  @ApiOperation({
+    summary: 'Get NearBy Friend',
+  })
+  @ApiOkResponse({
+    type: GetUserNearByFriendResponse,
+  })
+  @ApiBadRequestResponse(SwaggerBadRequestResponse())
+  @Get('/:id/nearby')
+  async get(@Param('id', new ObjectIdValidationPipe()) userId: string) {
```

Comment:
if a user gets tons of nearby users, I see there is a potential performance issue for this API? do you have an idea how to improve it?

Reply:
* User information can be obtained concurrently
or
* Write the basic information directly to the member of Geo Zset